### PR TITLE
Localize all strings in FidesJS banner & modal based on the user's locale and experience configuration

### PR DESCRIPTION
**Pull request by NevilleS**
### Description Of Changes

This (should!) fully localize the FidesJS banner & modal components. There's some additional follow-up work to do for testing and sending the right history IDs based on the user locale, but this should cover most of the UI work!

### Code Changes

- [X] Update all fides.js components to use `i18n.t(...)` translations instead of static strings or `experience` properties
- [X] Add `messageExists` helper function for cases where we want to have optional labels
- [X] Add E2E tests for optional GPC UI components
- [X] Add E2E tests for optional privacy policy link components
- [X] Add E2E tests for optional banner title/description overrides
- [X] Add static strings for `es` locale
- [X] Disable automatic locale detection if the `privacy-experience` API has it disabled

Remaining items that I'll roll into the next PR for PROD-1597:
- [ ] Add E2E tests that cover the special case where an experience is configured with a translation, but there is not a matching translation for all notices
- [ ] Run Typescript linter on fides-js (there are type errors not being surfaced by eslint)
- [ ] Fix other TODOs for PROD-1597 👍 

Follow-up on (PROD-1744) to fixing saving preferences and notices served:
- [ ] Ensure the correct `privacy_experience_config_history_id` and `privacy_notice_history_id` values are used for reporting based on the selected locale
- [ ] Re-enable disabled tests (marked with `TODO (PROD-1744)`)

Follow-up (PROD-1598):
- [ ] Add ability for user to manually select from available locales in banner & modal

### Pre-Merge Checklist

* [X] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [X] Issue Requirements are Met
* [X] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/fides-js/src/components/ConsentBanner.tsx [hunk 1]
```diff
@@ -94,10 +95,7 @@ const ConsentBanner: FunctionComponent<BannerProps> = ({
                   {bannerTitle}
                 </div>
                 {showGpcBadge && (
-                  <GpcBadge
-                    label="Global Privacy Control Signal"
-                    status="detected"
-                  />
+                  <GpcBadge i18n={i18n} status={GpcStatus.APPLIED} />
```
Comment (NevilleS) [inline clients/fides-js/src/components/ConsentBanner.tsx:98]: Modified this _slightly_ to not invent a new "Detected" status type and reuse the same translation.

File: clients/fides-js/src/lib/i18n/i18n-utils.ts [hunk 2]
```diff
@@ -265,6 +265,16 @@ export function matchAvailableLocales(
...
 }
 
+/**
+ * Check if the given message exists in the current locale's catalog.
+ */
+export function messageExists(i18n: I18n, id: string): boolean {
+  // NOTE: Our i18n library mirrors LinguiJS's implementation, which doesn't
+  // expose a way to check for missing messages directly. However, it will
+  // always return the "id" if no matching message exists, so we rely on that!
+  return i18n.t(id) !== "" && i18n.t(id) !== id;
```
Comment (eastandwestwind) [inline clients/fides-js/src/lib/i18n/i18n-utils.ts:275]: The `id` we pass into this method is the locale id?

File: clients/privacy-center/cypress/support/commands.ts [hunk 3]
```diff
@@ -187,7 +191,7 @@ declare global {
        */
       visitConsentDemo(
         options?: FidesConfig,
-        queryParams?: any,
+        queryParams?: Cypress.VisitOptions["qs"],
```
Comment (eastandwestwind) [inline clients/privacy-center/cypress/support/commands.ts:194]: 💯

**Review by eastandwestwind — APPROVED** [review]
Looking good, let's get this integrated and continue the testing!

File: clients/fides-js/src/components/notices/NoticeToggles.tsx [hunk 4]
```diff
@@ -30,21 +33,26 @@ const NoticeToggles = ({
   return (
     <div>
       {notices.map((notice, idx) => {
+        const i18nKey = `exp.notices.${notice.id}`;
+        const title = i18n.t(`${i18nKey}.title`);
+        const description = i18n.t(`${i18nKey}.description`);
         const checked = enabledNoticeKeys.indexOf(notice.notice_key) !== -1;
         const isLast = idx === notices.length - 1;
-        const dataUse = { key: notice.notice_key, name: notice.name };
+        // TODO: check
```
[the remark this answers is not in the document]
Comment (NevilleS) [reply clients/fides-js/src/components/notices/NoticeToggles.tsx:41]: Hmm usually my TODO notes-to-self are more instructive than this one...!

File: clients/privacy-center/cypress/support/commands.ts (hunk 3 above)
Comment (NevilleS) [reply clients/privacy-center/cypress/support/commands.ts:194]: > "Leave this world better a little better than you found it." - Robert Baden-Powell

File: clients/fides-js/src/lib/i18n/i18n-utils.ts (hunk 2 above)
Comment (NevilleS) [reply clients/fides-js/src/lib/i18n/i18n-utils.ts:275]: It's the message id - so things like `exp.title`, or `static.gpc.status.applied`

File: clients/fides-js/src/components/notices/NoticeToggles.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (NevilleS) [reply clients/fides-js/src/components/notices/NoticeToggles.tsx:41]: Right - I changed this to not bundle things together into `DataUse` and instead pass those props separately to as `title` & `noticeKey` instead. I tested this 👍

**Pull request merged**